packages/liferay-js-insights/insights/dependencies.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/**
+ * Extracts import declarations from a given AST applying the following transforms:
+ * - Relative imports are discarded
+ * - Imports are grouped per origin into:
+ * 	- clay3
+ *  - react
+ *  - js
+ *  - metal
+ *  - clay2
+ *  - others
+ */
+function extractImports(ast) {
+	const EMPTY = [[], [], [], [], [], []];
+	const SOURCE_GROUPS = [
+		source => source.includes('@clayui'),
+		source =>
+			source.includes('react') ||
+			source.includes('prop-types') ||
+			source.includes('classnames'),
+		source => source.includes('frontend-js-web'),
+		source => source.includes('metal-') || source === 'metal',
+		source => source.includes('clay-') || source === 'clay',
+		_ => true // eslint-disable-line
+	];
+
+	function getGroup(source) {
+		return SOURCE_GROUPS.findIndex(group => group(source));
+	}
+
+	const externalImports = ast.program.body.filter(
+		node =>
+			node.type === 'ImportDeclaration' &&
+			!node.source.value.includes('./')
```

Comment:
I think you want `startsWith` here too, but seeing as you also need to check fo `'../'` (and arguably leading `/` as well, although I have a lint to stop those) maybe you should just use a regex. It could be something as simple as `/^[a-z@]/i.test(node.source.value)`.

Reply:
👍